addons/mail/static/src/models/emoji_view.js

```diff
@@ -30,7 +30,8 @@ registerModel({
             if (!this.exists()) {
                 return;
             }
-            this.update({ emojiGridViewAsHovered: this.emojiGridItemViewOwner.emojiGridRowViewOwner.emojiGridViewOwner });
+            this.update({ emojiPickerViewAsHovered: this.emojiPickerViewOwner });
+            this.emojiPickerViewAsHovered.emojiSearchBarView.update({ placeholder: this.emoji.shortcodes});
```

Comment:
`placeholder` is a declarative field, therefore the update of this field should stay declarative, not be manually and imperatively updated.

So the idea consists of changing the compute method of `placeholder` so that its value is computed based on `hoveredEmojiPickerView`.

Reply:
BTW it works to imperatively update the field, but it's very prone to race conditions in the future.
It's very desirable to have as many declarative fields as possible, and when it's declarative it's important for it to stay declaratively updated.

Right now code allows imperative update on these fields just because there are still some problematic occurrences to fix, and we don't want to raise a traceback for users. However, we will eventually disable this possibility in the future, when we are sure that all of them are converted.